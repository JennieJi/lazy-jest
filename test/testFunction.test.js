const { testFunction } = require('../src/testFunction');

describe('empty arg function', () => {
  const emptyArgFunc = () => 'Success!';

  testFunction(emptyArgFunc);
  testFunction(emptyArgFunc, [], 'custom test caption');
});

describe('function with args', () => {
  const multiArgsFunc = (a, b) => a + b;

  testFunction(multiArgsFunc, []);
  testFunction(multiArgsFunc, [
    [null, 1],
    [NaN],
    [0, 1],
    ['0', 1]
  ]);
});

describe('function will throw', () => {
  const funcWillThrow = () => {
    throw 'Error!';
  }

  testFunction(funcWillThrow);
});