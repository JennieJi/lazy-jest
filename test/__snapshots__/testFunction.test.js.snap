// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty arg function custom test caption - () 1`] = `"Success!"`;

exports[`empty arg function emptyArgFunc() - () 1`] = `"Success!"`;

exports[`function will throw funcWillThrow() - () 1`] = `"Error!"`;

exports[`function with args multiArgsFunc() - ("0",1) 1`] = `"01"`;

exports[`function with args multiArgsFunc() - () 1`] = `NaN`;

exports[`function with args multiArgsFunc() - (0,1) 1`] = `1`;

exports[`function with args multiArgsFunc() - (NaN) 1`] = `NaN`;

exports[`function with args multiArgsFunc() - (null,1) 1`] = `1`;
